command/agent/check.go

```diff
@@ -532,6 +545,33 @@ func (c *CheckTCP) check() {
 	c.Notify.UpdateCheck(c.CheckID, structs.HealthPassing, fmt.Sprintf("TCP connect %s: Success", c.TCP))
 }
 
+// checkUDP is invoked periodically to perform the UDP check
+func (c *CheckTCP) checkUDP() {
+	conn, err := c.dialer.Dial(`udp`, c.UDP)
+	if err == nil {
+		b := []byte("CONNECTED-MODE SOCKET")
```

Comment:
Looking at the other UDP checks there's usually control of the payload sent and a check of the payload received. We should probably add that here so people get a little control over how the check works.